### PR TITLE
feat(air): migrate range checker bus to LogUp

### PR DESCRIPTION
**Pull request by grjte**
This PR addresses #982 and #299 (no longer needed), simplifying the range checker to require only 2 main trace columns and a single auxiliary trace column.

- [x] rewrite AIR constraints
- [x] rewrite trace generation of main range checker trace
- [x] rewrite auxiliary trace generation for range checker lookup bus
- [x] update tests

### CODE REVIEW

**Comment by grjte** [comment]
Notes:
- the docs need a really big update, so I think I should tackle it in a separate PR
- Currently we pad the range checker at the beginning of the trace. I think we could now pad at the end with rows of u16::MAX of multiplicity 0, and this would simplify things. Wdyt? I left it as is for now, to reduce the amount of changes.

File: air/src/constraints/range.rs [hunk 1]
```diff
@@ -113,48 +118,39 @@ pub fn enforce_aux_constraints<F, E>(
...
+/// for which lookups are requested from the stack and memory are each looked up with multiplicity
+/// one, and the logarithmic derivatives are subtracted from b_range.
+///
+/// Let the following variables:
+/// - rc_value: the range checker value
+/// - rc_multiplicity: the range checker multiplicity value
+/// - flag_s: boolean flag indicating a stack operation with range checks
+/// - sv0-sv3: stack value 0-3, the 4 values range-checked from the stack
+/// - flag_m: boolean flag indicating the memory chiplet is active (i.e. range checks are required)
+/// - mv0-mv1: memory value 0-1, the 2 values range-checked from the memory chiplet
```
Comment (bobbinth) [inline air/src/constraints/range.rs:141]: nit: one thing that might be helpful to note here is the degrees of `flag_s` and `flag_m`. I believe these are 3 for both - but I had look in a couple of places to figure it out.

File: processor/src/range/aux_trace.rs [hunk 2]
```diff
@@ -29,102 +22,84 @@ impl AuxTraceBuilder {
...
 
         // after the padded section of the range checker table, include `z` in the running product
         // at each step and remove lookups from user ops at any step where user ops were executed.
-        for (row_idx, (hint, lookup)) in self
-            .row_flags
+        for (row_idx, (multiplicity, lookup)) in main_trace
+            .get_column(M_COL_IDX)
             .iter()
             .zip(main_trace.get_column(V_COL_IDX).iter())
             .enumerate()
```
Comment (bobbinth) [inline processor/src/range/aux_trace.rs:105]: nit: we should probably change "running product" to "running sum" in the comments.

**Review by bobbinth — APPROVED** [review]
Looks great! Thank you! I left a few small blocking comments inline - some of them can be left for future PRs.

Re moving padding to the end - yes, let's leave it for a future PR. We can also combine this with one other potential optimization idea i left inline.

**Comment by bobbinth** [comment]
Oh - one other thing: let's update the changelog to include this change.

**Pull request merged**